bundle/Command/UpdateContentVisibilityCommand.php

```diff
@@ -0,0 +1,261 @@
+                        $contentId,
+                        [$language->getLanguageCode()],
+                    ),
+                );
+            } catch (NotFoundException $exception) {
+                $this->logger->error(
+                    sprintf(
+                        'Content with id #%d does not exist: %s',
+                        $contentId,
+                        $exception->getMessage(),
+                    ),
+                );
+
+                continue;
+            }
+
+            try {
+                $action = $this->scheduledVisibilityService->updateVisibilityIfNeeded($content);
+            } catch (InvalidStateException $exception) {
+                $this->logger->error($exception->getMessage());
+
+                continue;
+            }
+
+            if ($action !== VisibilityUpdateResult::NoChange) {
+                $this->logger->info(
+                    sprintf(
+                        "Content '%s' with id #%d has been %s.",
+                        $content->getName(),
+                        $content->getId(),
+                        $action->value,
+                    ),
+                );
+            }
+
+            $this->style->progressAdvance();
+        }
+    }
+
+    private function getQueryBuilder(): QueryBuilder
+    {
+        $query = $this->connection->createQueryBuilder();
+        $query
+            ->select('id', 'initial_language_id')
+            ->from('ezcontentobject')
+            ->where('published != :unpublished')
+            ->orderBy('id', 'ASC')
+            ->setParameter('unpublished', 0);
+
+        return $query;
+    }
+
+    private function applyContentTypeLimit(QueryBuilder $query, array $contentTypeIds): void
+    {
+        $query->where(
+            $query->expr()->in('contentclass_id', ':content_type_ids'),
+        )->setParameter('content_type_ids', $contentTypeIds, Connection::PARAM_INT_ARRAY);
+    }
+
+    private function getPager(QueryBuilder $query, array $contentTypeIds): Pagerfanta
+    {
+        $countQueryBuilderModifier = static function (QueryBuilder $queryBuilder) use ($contentTypeIds): void {
+            $queryBuilder->select('COUNT(id) AS total_results')
+                ->from('ezcontentobject')
+                ->where('published != :unpublished')
+                ->setParameter('unpublished', 0)
+                ->setMaxResults(1);
+
+            if (count($contentTypeIds) > 0) {
+                $queryBuilder->where(
+                    $queryBuilder->expr()->in('contentclass_id', ':content_type_ids'),
+                )->setParameter('content_type_ids', $contentTypeIds, Connection::PARAM_INT_ARRAY);
+            }
+        };
+
+        return new Pagerfanta(new QueryAdapter($query, $countQueryBuilderModifier));
+    }
+
+    private function loadLanguage(int $id, InputInterface $input): Language
+    {
+        $cacheItem = $this->cache->getItem("netgen-ibexa-scheduled-visibility-language-{$id}");
```

Comment:
This seems like an overkill TBH, in-memory caching through a private property would be perfectly fine. So something like:

```php
if (!isset($this->languageCache[$id])) {
    $language = $this->repository->getContentLanguageService()->loadLanguageById($id);
    $this->languageCache[$id] = $language;
}

return $this->languageCache[$id];
```

Reply:
fixed in 236df91cc6c8ccc6ff59dd26aef8308b79de5275